library_generation/new_client/new-client.py

```diff
@@ -0,0 +1,458 @@
+            "generation, a new API version of an existing module does not "
+            "require new-client.py invocation. "
+            "See go/yoshi-java-new-client#adding-a-new-service-version-by-owlbot."
+        )
+    print(f"Creating a new module {workdir}")
+    os.makedirs(workdir, exist_ok=False)
+    # write .repo-metadata.json file
+    with open(workdir / ".repo-metadata.json", "w") as fp:
+        json.dump(repo_metadata, fp, indent=2)
+
+    # create owlbot.py
+    templates.render(
+        template_name="owlbot.py.j2",
+        output_name=str(workdir / "owlbot.py"),
+        should_include_templates=True,
+        template_excludes=[],
+    )
+
+    # In monorepo, .OwlBot.yaml needs to be in the directory of the module.
+    owlbot_yaml_location_from_module = ".OwlBot.yaml"
+    # create owlbot config
+    templates.render(
+        template_name="owlbot.yaml.monorepo.j2",
+        output_name=str(workdir / owlbot_yaml_location_from_module),
+        artifact_name=distribution_name_short,
+        proto_path=proto_path,
+        module_name=f"java-{api_shortname}",
+        api_shortname=api_shortname
+    )
+
+    print(f"Pulling proto from {googleapis_url}")
+    output_dir = Path(f"{sys.path[0]}/../../output").resolve()
+    __sparse_clone(
+        remote_url=googleapis_url,
+        dest=output_dir,
+    )
+    # Find a versioned directory within proto_path
+    # We only need to generate one version of the library as OwlBot
+    # will copy other versions from googleapis-gen.
+    version = __find_version(
+        Path(f"{sys.path[0]}/../../output/{proto_path}").resolve()
+    )
+    versioned_proto_path = f"{proto_path}/{version}"
+    print(f"Generating from {versioned_proto_path}")
+    # parse BUILD.bazel in proto_path
+    client_input = parse(
+        build_path=Path(f"{sys.path[0]}/../../output/{versioned_proto_path}")
+        .resolve(),
+        versioned_path=versioned_proto_path,
+    )
+    repo_root_dir = Path(f"{sys.path[0]}/../../").resolve()
+    # run generate_library.sh
+    subprocess.check_call([
+        "library_generation/generate_library.sh",
+        "-p",
+        versioned_proto_path,
+        "-d",
+        destination_path,
+        "--gapic_generator_version",
+        generator_version,
+        "--protobuf_version",
+        "23.2",
+        "--proto_only",
+        client_input.proto_only,
+        "--gapic_additional_protos",
+        client_input.additional_protos,
+        "--transport",
+        client_input.transport,
+        "--rest_numeric_enums",
+        client_input.rest_numeric_enum,
+        "--gapic_yaml",
+        client_input.gapic_yaml,
+        "--service_config",
+        client_input.service_config,
+        "--service_yaml",
+        client_input.service_yaml,
+        "--include_samples",
+        client_input.include_samples,
+        "--versions_file",
+        f"{repo_root_dir}/versions.txt"],
+        cwd=repo_root_dir
```

Comment:
After @diegomarquezp 's PR is in, the logic here needs to be updated to apply generate_library.sh and postprocess_lirbary separately.